test/spmd/test_tracing.py

```diff
@@ -33,13 +34,15 @@ def _test_tracing_all_reduce_nd(self, mesh_tensor):
             ]
 
             def fn(tensor: torch.Tensor):
-                reduced_tensor = mesh.all_reduce(tensor, mesh_dim=dim)
+                # tensor_to_reduce = tensor.clone().contiguous()
+                mesh.all_reduce(CommTensor(tensor), mesh_dim=dim)
                 # multiply with 1 to trigger wait on read during tracing.
-                return reduced_tensor * 1
+                return tensor * 1
 
             # use a local_tensor + 1 for tracing to make sure that we are not
             # simply replaying recorded tensor value
-            traced_fn = make_fx(fn)(local_tensor + 1)
+            traced_fn = make_fx(fn)(local_tensor.clone() + 1)
+            print(f">>>> traced graph: {traced_fn}")
```

Comment:
what's the current issue?

Reply:
I actually want to see if we can move CommTensor wrapping to `redistribute_local_tensor` level, so trying things out right now, and it currently hit issues like there're multiple global tracer while CommTensor asserting it should only have one.

```
E             File "/Users/wanchaol/conda/envs/ptdev/lib/python3.8/site-packages/torch/utils/_pytree.py", line 192, in tree_map
E               return tree_unflatten([fn(i) for i in flat_args], spec)
E             File "/Users/wanchaol/conda/envs/ptdev/lib/python3.8/site-packages/torch/utils/_pytree.py", line 192, in <listcomp>
E               return tree_unflatten([fn(i) for i in flat_args], spec)
E             File "/Users/wanchaol/Documents/tau/spmd/api.py", line 45, in redistribute
E               comm_arg = CommTensor(arg)
E             File "/Users/wanchaol/conda/envs/ptdev/lib/python3.8/site-packages/torch/distributed/_spmd/comm_tensor.py", line 105, in __new__
E               if _get_tracer(t) is None:
E             File "/Users/wanchaol/conda/envs/ptdev/lib/python3.8/site-packages/torch/distributed/_spmd/comm_tensor.py", line 59, in _get_tracer
E               assert len(keys) == 1
E           AssertionError
```